internal/hook/item_hook_handler.go

```diff
@@ -482,11 +478,15 @@ func GroupRestoreExecHooks(
 	}
 	hookFromAnnotation := getPodExecRestoreHookFromAnnotations(metadata.GetAnnotations(), log)
 	if hookFromAnnotation != nil {
-		byContainer[hookFromAnnotation.Container] = []NamedExecRestoreHook{
+		// default to first container in pod if unset
+		if hookFromAnnotation.Container == "" {
+			hookFromAnnotation.Container = pod.Spec.Containers[0].Name
+		}
+		byContainer[hookFromAnnotation.Container] = []PodExecRestoreHook{
 			{
-				Name:   "<from-annotation>",
-				Source: "annotation",
-				Hook:   *hookFromAnnotation,
+				HookName:   "<from-annotation>",
```

Comment:
`"<from-annotation>"` still needs to be substituted with the real thing

Reply:
If we add an annotation for hook name it will deviate from the behavior of [backup hooks](https://github.com/vmware-tanzu/velero/blob/main/internal/hook/item_hook_handler.go#L205). Is that ok?